assets/blocks/quiz/category-question-block/category-question-edit.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * WordPress dependencies
+ */
+import { __, _n, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useQuestionNumber } from '../question-number';
+import { useQuestionCategories } from '../question-categories';
+import CategoryQuestionSettings from './category-question-settings';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Quiz category question block editor.
+ *
+ * @param {Object}   props
+ * @param {Object}   props.attributes              Block attributes.
+ * @param {Object}   props.attributes.categoryName Category name.
+ * @param {Function} props.setAttributes           Set block attributes.
+ */
+const CategoryQuestionEdit = ( props ) => {
+	const {
+		attributes: {
+			options: { number = 1, category },
+		},
+		clientId,
+		setAttributes,
+	} = props;
+	const questionNumber = useQuestionNumber( clientId );
+	const [ , getCategoryTermById ] = useQuestionCategories();
+
+	const range =
+		1 === number
+			? questionNumber
+			: `${ questionNumber } - ${ questionNumber + number - 1 }`;
+
+	const questionIndex = (
+		<h2 className="sensei-lms-question-block__index">{ range }.</h2>
+	);
+
+	const categoryName =
+		getCategoryTermById( category )?.name ?? props.attributes.categoryName;
+	const categoryNameMatch = categoryName === props.attributes.categoryName;
+
+	useEffect( () => {
+		if ( categoryName && ! categoryNameMatch ) {
+			setAttributes( {
+				categoryName,
+			} );
+		}
+	}, [ categoryName, categoryNameMatch, setAttributes ] );
+
+	return (
+		<>
+			<CategoryQuestionSettings { ...props } />
+			<div
+				className={ `sensei-lms-question-block sensei-lms-category-question-block ${
```

Comment:
I think we're not doing anything with the `sensei-lms-category-question-block` class, right?

Reply:
Nope, just setting up the strategy for when we do want to.